src/main/java/com/microsoft/spring/data/gremlin/repository/support/GremlinEntityInformation.java

```diff
@@ -23,19 +25,29 @@
     private Field id;
     private String entityLabel;
     private GremlinEntityType entityType;
+    private GremlinSource gremlinSource;
 
-    public GremlinEntityInformation(Class<T> domainClass) {
+    public GremlinEntityInformation(@NonNull Class<T> domainClass) {
```

Comment:
seems there's a `@notempty` annotation used in azure-spring-boot, you may want to be consistent

Reply:
@NotEmpty is for String, Array, map and Collection, include null and not empty. May not very suitable here.